.github/workflows/ci.yml

```diff
@@ -1,24 +1,156 @@
+
+      # - name: cargo clippy (no default features) (auxiliary)
+      #   run: cargo clippy --workspace --tests --benches --examples --no-default-features -- -D warnings
+
+      - name: cargo clippy (default features)
+        run: cargo clippy --workspace --lib --bins -- -D warnings
+
+      - name: cargo clippy (default features) (auxiliary)
+        run: cargo clippy --workspace --tests --benches --examples -- -D warnings
+
+      # As above
+      # - name: cargo clippy (all features)
+      #   run: cargo clippy --workspace --lib --bins --all-features -- -D warnings
+
+      # - name: cargo clippy (all features) (auxiliary)
+      #   run: cargo clippy --workspace --tests --benches --examples --all-features -- -D warnings
+
+      # At the time of writing, we don't have any tests. Nevertheless, it's better to still run this
+      - name: cargo test
+        run: cargo test --workspace --all-features
+  
+  clippy-msrv:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # We don't have any platform specific code, so only run this job once
+        os: [ubuntu-latest]
+    name: cargo clippy (minimum supported rust version)
+    env:
+      RUSTFLAGS: "--cfg peniko_msrv"
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: install msrv toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.MINIMUM_SUPPORTED_RUST_VERSION }}
+          components: clippy
+
+      # We don't currently have features, so there's no need to run these checks
+      # - name: cargo clippy (no default features)
+      #   run: cargo clippy --workspace --lib --bins --no-default-features -- -D warnings
+
+      # - name: cargo clippy (no default features) (auxiliary)
+      #   run: cargo clippy --workspace --tests --benches --examples --no-default-features -- -D warnings
+
+      - name: cargo clippy (default features)
+        run: cargo clippy --workspace --lib --bins -- -D warnings -A clippy::doc_markdown
+
+      - name: cargo clippy (default features) (auxiliary)
+        run: cargo clippy --workspace --tests --benches --examples -- -D warnings -A clippy::doc_markdown
+
+      # As above
+      # - name: cargo clippy (all features)
+      #   run: cargo clippy --workspace --lib --bins --all-features -- -D warnings
+
+      # - name: cargo clippy (all features) (auxiliary)
+      #   run: cargo clippy --workspace --tests --benches --examples --all-features -- -D warnings
+
+  clippy-stable-wasm:
+    runs-on: ubuntu-latest
+    name: cargo test (wasm32)
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: install stable toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_STABLE_VER }}
+          targets: wasm32-unknown-unknown
+          components: clippy
+
+      - name: cargo clippy (wasm)
+        run: cargo clippy --all-targets --target wasm32-unknown-unknown -- -D warnings
```

Comment:
Is `--all-targets` necessary here (when it is already done in other jobs?)?

Reply:
I think so.

This ensures that any tests, benchmarks and examples check for wasm (even though there currently are none of those)

This is cargo having two different meanings of the word "target" (or well at least three, if you include the target directory).